<a href="https://colab.research.google.com/github/lalitpandey02/EDA-on-bestselling-books/blob/main/Best_Booksellers_Dataset.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### **Categorical Feature Analysis**
- Univariate analysis - Line Chart, Bar Plot and Histogram
- Bivariate Analysis - Scatter Plot, Regression Plot, Stacked Bar, Bivariate line chart
- Multivariate Analysis - Scatter Plot, Regression Plot, Stacked Bar, Bivariate line chart, Heat Map

### **Numerical Feature Analysis**
- Correlation - Scatter Plot and Regression Plot
- Trend - Line Chart , Bar plot
- Distribution - Bar Plot and Pie Charts

## **EDA of Best Book Sellers  Dataset**

---------Questions--------------
1. Authors with highest user rating
2. Book name with highest and lowest number of reviews
3. Book with highest and lowest price
4. Year with highest and lowest number of sales
5. count of fiction and non-fiction books
6. Categorize User Rating
7. Price comparision between Non-Fiction and Fiction
8. Book Prices with lowest and highest sales

## **Study the dataset** 

In [1]:
import numpy as np 
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns 
import plotly.express as px
%matplotlib inline 

In [2]:
url = "https://raw.githubusercontent.com/lalitpandey02/EDA-on-bestselling-books/main/bestsellers%20with%20categories.csv"
df = pd.read_csv(url)

In [3]:
df.head()

Name  \
0                      10-Day Green Smoothie Cleanse   
1                                  11/22/63: A Novel   
2            12 Rules for Life: An Antidote to Chaos   
3                             1984 (Signet Classics)   
4  5,000 Awesome Facts (About Everything!) (Natio...   

                     Author  User Rating  Reviews  Price  Year        Genre  
0                  JJ Smith          4.7    17350      8  2016  Non Fiction  
1              Stephen King          4.6     2052     22  2011      Fiction  
2        Jordan B. Peterson          4.7    18979     15  2018  Non Fiction  
3             George Orwell          4.7    21424      6  2017      Fiction  
4  National Geographic Kids          4.8     7665     12  2019  Non Fiction

In [4]:
print(df.shape)
df.info()

(550, 7)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 550 entries, 0 to 549
Data columns (total 7 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Name         550 non-null    object 
 1   Author       550 non-null    object 
 2   User Rating  550 non-null    float64
 3   Reviews      550 non-null    int64  
 4   Price        550 non-null    int64  
 5   Year         550 non-null    int64  
 6   Genre        550 non-null    object 
dtypes: float64(1), int64(3), object(3)
memory usage: 30.2+ KB


In [5]:
df.isnull().sum()

Name           0
Author         0
User Rating    0
Reviews        0
Price          0
Year           0
Genre          0
dtype: int64

In [6]:
df.describe()

User Rating       Reviews       Price         Year
count   550.000000    550.000000  550.000000   550.000000
mean      4.618364  11953.281818   13.100000  2014.000000
std       0.226980  11731.132017   10.842262     3.165156
min       3.300000     37.000000    0.000000  2009.000000
25%       4.500000   4058.000000    7.000000  2011.000000
50%       4.700000   8580.000000   11.000000  2014.000000
75%       4.800000  17253.250000   16.000000  2017.000000
max       4.900000  87841.000000  105.000000  2019.000000

Observations:
- There are no missing records
- Contains books from the year 2009 to 2019
- Some books have a price of 0 so those could be complementary or free books available for all
- Mean price, and Q1 , Q2 , Q3 Quartiles are in the range of 11 to 16 but the Max price is 105. So there seems to be some outliers 

#### **Variable types in the dataset**

In [7]:
df.columns

Index(['Name', 'Author', 'User Rating', 'Reviews', 'Price', 'Year', 'Genre'], dtype='object')

- Categorical Variables : Genre , Year
- Numerical Variables : User Rating, Reviews, Price

#### **Checking Outliers**

Price is the primary column which seems to be containing outliers

Distribution of Price

In [8]:
px.histogram(x = df.Price)

In [9]:
px.scatter(data_frame=df, y = "Price" , color = "Genre")

In [10]:
px.box(df, x = "Genre", y = "Price", color = "Genre")

As we see in the box plot and the scatter plot, the price of the books are dense in region below 60. Hence we can drop the records with price above 60

In [11]:
# df1 = df.drop()